cmake/external/icu.cmake

```diff
@@ -0,0 +1,164 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+include(FetchContent)
+
+set(ICU_VERSION "70")
+set(ICU_TARGET_NAME "icu_external")
+set(ICU_URL https://github.com/unicode-org/icu/releases/download/release-70-1/icu4c-70_1-src.tgz)
+set(ICU_URL_HASH SHA256=8d205428c17bf13bb535300669ed28b338a157b1c01ae66d31d0d3e2d47c3fd5)
+
+set(THIRD_PARTY_PATH ${CMAKE_BINARY_DIR}/_deps/icu)
+set(ICU_SOURCE_DIR  ${THIRD_PARTY_PATH}/icu-src)
+set(ICU_INSTALL_DIR ${THIRD_PARTY_PATH}/icu-install)
+
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -std=c++11")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+endif()
+
+if(OPENVINO_RUNTIME_COMPILE_DEFINITIONS)
+    foreach(def ${OPENVINO_RUNTIME_COMPILE_DEFINITIONS})
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D${def}")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D${def}")
+    endforeach()
+endif()
+
+set(HOST_ENV_CMAKE ${CMAKE_COMMAND} -E env
+        CFLAGS=${CMAKE_C_FLAGS}
+        CXXFLAGS=${CMAKE_CXX_FLAGS}
+        LDFLAGS=${CMAKE_MODULE_LINKER_FLAGS}
+)
+
+# ICU supports only Release and Debug build types 
+if(GENERATOR_IS_MULTI_CONFIG_VAR)
+  set(ICU_CONFIGURE_FLAGS $<$<CONFIG:Debug>:"--enable-debug">$<$<NOT:$<CONFIG:Release>>:"--enable-release">)
+  set(ICU_BUILD_TYPE $<$<CONFIG:Debug>:Debug>$<$<NOT:$<CONFIG:Debug>>:Release>)
```

Comment:
in `else` block, build types are written with capital letters, is a case important?

Reply:
you are right it affects Ninja Multi-Config build on Linux, it is broken by sentence peace now, but I've fixed it for icu